src/client/src/commons/form/profile/components/profileLayers/components/chronoStratigraphyLayers/layerCard.js

```diff
@@ -0,0 +1,346 @@
+import React, { useEffect, useState, useCallback } from "react";
+import {
+  CardActionArea,
+  Stack,
+  Typography,
+  Box,
+  Card,
+  IconButton,
+  TextField,
+  InputAdornment,
+  Skeleton,
+  TableCell,
+  Table,
+  Tooltip,
+  TableRow,
+  CircularProgress,
+  Autocomplete,
+  TableBody,
+} from "@mui/material";
+import { Close, Delete, Edit } from "@mui/icons-material";
+import { ClickAwayListener } from "@mui/base";
+import { useTranslation } from "react-i18next";
+import {
+  useChronostratigraphyMutations,
+  useDomainSchema,
+} from "../../../../../../../api/fetchApiV2";
+
+const State = Object.freeze({
+  EDITING: Symbol("Editing"),
+  EDITABLE: Symbol("Editable"), // borehole editing session started but not yet editing this layer
+  DELETED: Symbol("Deleted"),
+  DISPLAY: Symbol("Display"),
+});
+
+const LayerCard = ({
+  layer, // {id, fromDepth, toDepth, chronostratigraphyId}: the layer object to display
+  minFromDepth, // number: minimal allowed fromDepth
+  maxToDepth, // number: maximal allowed toDepth
+  header, // Array<{title, isVisible}>: header object with titles and visibility
+  isEditable, // boolean: specifies if values can be edited
+  height, // height of the layerCard
+}) => {
+  const { t, i18n } = useTranslation();
+
+  const {
+    update: { mutate: updateChronostratigraphy },
+    delete: { mutate: deleteChronostratigraphy },
+  } = useChronostratigraphyMutations();
+  const { data: schemaData } = useDomainSchema(
+    "custom.chronostratigraphy_top_bedrock",
+  );
+
+  const [fromDepthErrorMessage, setFromDepthErrorMessage] = useState("");
+  const [toDepthErrorMessage, setToDepthErrorMessage] = useState("");
+  const [fromDepth, setFromDepth] = useState(null);
+  const [toDepth, setToDepth] = useState(null);
+  const [options, setOptions] = useState(null);
+  const [selection, setSelection] = useState(null);
+  const [cardState, setCardState] = useState(
+    isEditable ? State.EDITABLE : State.DISPLAY,
+  );
+
+  // create options array from codelist schema
+  useEffect(() => {
+    if (schemaData) {
+      setOptions(
+        [...schemaData]
+          .sort((a, b) => a.order - b.order)
+          .reduce((accu, d) => {
+            const level = d.path.split(".").length - 1;
```

Comment:
Q: was hat das mit dem split nach "." auf sich?

Reply:
Die string representation vom Pfad (typ [LTree](https://www.postgresql.org/docs/current/ltree.html)) verwendet einen Punkt zum abtrennen der Pfad-Elemente. z.B. `15001001.15001049.15001065.15001070.15001168.15001073`. Hier wird gezählt wie viele Punkte `.` es im Pfad hat um schlussendlich herauszufinden in welchem dropdown die option angezeigt werden muss.